### ISSUE TITLE
You don't seem to understand Clean Architecture

### ISSUE THREAD

**Issue by mycroftcanner** [sergdort/CleanArchitectureRxSwift#72]
Your domain has entities... but they have absolutely no logic or critical business rules associated to them... it only defines entities and a use case provider protocol...

You have 3 redundant use cases provider implementations that are fully aware of everything and can't talk to each other.

Use cases implementation should have no knowledge about core data, realm, etc... you just write platform agnostic code... using protocols and inject adapters as dependencies.

**Comment by mycroftcanner** [comment]
CleanCodeRxSwift would be a better name for this repo. 

You write clean code I give you that! But calling this clean architecture is a misnomer and would only confuse people.

**Comment by sergdort** [comment]
> Your domain has entities... but they have absolutely no logic or critical business rules associated to them... it only defines entities and a use case provider protocol...

I agree, the reason why domain in this example is “empty” module with no implementation, is because I wanted to implement 3 different concrete implement of the same domain and put them into separate module. In real app I would definitely not do that, I’d have domain which hides implementation detains behind adstraction, what I mean that all concrete usecases would be there and accessed via Service locator or any other DI pattern.

> But calling this clean architecture is a misnomer and would only confuse people.

The reason I called it Clean Arch is because it’s literally implementation of what Uncle Bob was talking about here https://youtu.be/Nsjsiz2A9mg and it uses RxSwift 🤷‍♂️

**Comment by sergdort** [comment]
> You have 3 redundant use cases provider implementations that are fully aware of everything and can't talk to each other.

I don’t agree here, unless I’m missing something, all concrete use cases are internal and sit in separate modules, the only way to create them are via corresponding use cases providers

**Comment by dineshflock** [comment]
I see that efforts have been put to implement Uncle Bob's Clean arch, but I agree with @mycroftcanner that this is not clean architecture.

In clean arch we have domain that has implementations of uses cases along with interfaces of use cases.
Use case implementations(part of Domain layer) should never depend on DB as Domain should not depend on anything.

**Comment by sergdort** [comment]
> Use case implementations(part of Domain layer) should never depend on DB as Domain should not depend on anything

I’m struggle to understand how in this case the Domain is dependent on anything? It sits in separate module and does not know anything about the implementation of a DB

**Comment by BarryDuggan** [comment]
@mycroftcanner  
I'm learning about repository pattern in swift.
My understanding isnt good enough yet to understand this discussion fully.

Not to take away from the great work done in this repo but could you point me toward an example of what 'you' think repository pattern in clean architecture should look like? 
A link to an article or book on amazon would be great
